pkg/api/service/warnings.go

```diff
@@ -64,12 +64,11 @@ func GetWarningsForService(service, oldService *api.Service) []string {
 }
 
 func getWarningsForIP(fieldPath *field.Path, address string) []string {
-	// IPv4 addresses with leading zeros CVE-2021-29923 are not valid in golang since 1.17
-	// This will also warn about possible future changes on the golang std library
-	// xref: https://issues.k8s.io/108074
+	// Validation blocks new IPv4 addresses with leading 0s, but we can still warn
```

Comment:
nit: Validation blocks new IPv4 addresses that fail validation using netip.ParseAddr. In theory that is only addresses with leading zeroes, but that assumes our implementation up until this point has been bug free and no other value has ever been ingested into a real Kubernetes cluster.

The error message on the other hand is sufficiently expansive that I think it covers most cases